components/promise-component.js

```diff
@@ -0,0 +1,66 @@
+// promise-component.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+module.exports = function(def){
+    if (_.isEmpty(def.inPorts)) throw Error("Missing inPorts");
+    if (def.resolvePort && !_.isString(def.resolvePort.name))
```

Comment:
Are you sure you want these defined outside the promiseComponent?   If they were part of this component, you could skip the error checking here.